pkg/pillar/types/smarttypes.go

```diff
@@ -1,85 +1,132 @@
 /*
- * Copyright (c) 2020. Zededa, Inc.
+ * Copyright (c) 2022. Zededa, Inc.
  * SPDX-License-Identifier: Apache-2.0
  */
 
 package types
 
-//SmartData holds SMART details of the device
-type SmartData struct {
-	SmartCtl        SmartCtl       `json:"smartctl"`
-	Device          Device         `json:"device"`
-	SmartAttribute  SmartAttribute `json:"ata_smart_attributes"`
-	PowerOnTime     PowerOnTime    `json:"power_on_time"`
-	PowerCycleCount int64          `json:"power_cycle_count"`
-	Temperature     Temperature    `json:"temperature"`
-	RawData         string         `json:"raw_data"`
+//DeviceSmartInfo holds SMART details of the device
+type DeviceSmartInfo struct {
+	PowerOnTime     PowerOnTime `json:"power_on_time"`
+	PowerCycleCount int64       `json:"power_cycle_count"`
 }
 
-//Device holds device details
-type Device struct {
-	Name     string `json:"name"`
-	InfoName string `json:"info_name"`
-	Type     string `json:"type"`
-	Protocol string `json:"protocol"`
+//PowerOnTime of the disk
+type PowerOnTime struct {
+	Hours int64 `json:"hours"`
 }
 
-//SmartAttribute holds SMART attributes
-type SmartAttribute struct {
-	Revision int64   `json:"revision"`
-	Table    []Table `json:"table"`
+//NewSmartDataWithDefaults returns 'SmartData' with default values
+func NewSmartDataWithDefaults() *DeviceSmartInfo {
+	return &DeviceSmartInfo{
+		PowerOnTime:     PowerOnTime{Hours: -1},
+		PowerCycleCount: -1,
+	}
 }
 
-//Table schema for each SMART attribute
-type Table struct {
-	ID         int64  `json:"id"`
-	Name       string `json:"name"`
-	Value      int64  `json:"value"`
-	Worst      int64  `json:"worst"`
-	Thresh     int64  `json:"thresh"`
-	WhenFailed string `json:"when_failed"`
-	Flags      Flags  `json:"flags"`
-	Raw        Raw    `json:"raw"`
+// DiskType defines disk types
+type DiskType int
+
+// CollectingStatus specifies the state of the error collector
+type CollectingStatus int
+
+const (
+	// SmartScsiDiskType - SCSI disk
+	SmartScsiDiskType DiskType = iota
```

Comment:
I thought the normal approach was to have the value zero aka default be an unknown i.e., in this case have SmartDiskTypeUnknown DiskType = iota

Also, I think it is easier if the uniq part of the name (sata, nvme, etc) are at the end of the names.

Reply:
I agree, fixed